examples/example-bayesian-linear-regression.jl

```diff
@@ -13,7 +13,8 @@ using Statistics
 # Use the Soss probabilistic programming language
 # to define a Bayesian linear regression model:
 
-m = @model X,s,σ begin
+m = @model X,s,a begin
```

Comment:
Just to note, we could also have a prior on the noise scale. Also, this is valid Soss (unless I mess it up):
```julia
m = @model X, prior, noise begin
    n = size(X,1)
    k = size(X,2)
    β ~ prior |> iid(k)
    yhat = X * β
    y ~ For(n) do j
        noise(yhat[j])
    end
end
```

Reply:
Interesting. So the user provides the prior distributions as hyperparameters.

Let's try this out.

Reply:
Hmmm, wait in this case, how do you provide the prior for sigma?

Reply:
I.e. what would the user provide as `prior` and `noise` in this case?

Reply:
Ah, right. I was thinking something like
```julia
prior = Normal(0,10)
noise = @model μ begin
    σ ~ HalfNormal()
    y ~ Normal(μ, σ)
    return y
end
```

You can nest models like this, but return values aren't set up correctly yet. Maybe we just start simple for the proof of concept?

Reply:
> return values aren't set up correctly yet

Can you elaborate on this?